testcases/misc_tests/p11sak_test.sh

```diff
@@ -0,0 +1,419 @@
+#!/bin/bash
+#
+# COPYRIGHT (c) International Business Machines Corp. 2020
+#
+# This program is provided under the terms of the Common Public License,
+# version 1.0 (CPL-1.0). Any use, reproduction or distribution for this software
+# constitutes recipient's acceptance of CPL-1.0 terms which can be found
+# in the file LICENSE file or at https://opensource.org/licenses/cpl1.0.php
+
+# - The PKCS11_USER_PIN environment variable must hold the user pin.
+# - The SLOT environment variable must hold the slot id of the token under test.
+#
+# sudo -E ./p11sak_test.sh
+
+
+
+echo "** Now executing 'p11sak_test.sh'"
+
+
+# tmp files
+
+P11SAK_DES_PRE=p11sak-des-pre.out
+P11SAK_DES_POST=p11sak-des-post.out
+P11SAK_3DES_PRE=p11sak-3des-pre.out
+P11SAK_3DES_POST=p11sak-3des-post.out
+P11SAK_AES_PRE=p11sak-aes-pre.out
+P11SAK_AES_POST=p11sak-aes-post.out
+P11SAK_RSA_PRE=p11sak-rsa-pre.out
+P11SAK_RSA_POST=p11sak-rsa-post.out
+P11SAK_EC_PRE=p11sak-ec-pre.out
+P11SAK_EC_POST=p11sak-ec-post.out
+
+
+
+echo "** Now generating keys - 'p11sak_test.sh'"
+
+
+# generate objects
+
+# des
+p11sak generate-key des --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-des
+# 3des
+p11sak generate-key 3des --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-3des
+# aes [128 | 192 | 256]
+p11sak generate-key aes 128 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-aes-128
+p11sak generate-key aes 192 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-aes-192
+p11sak generate-key aes 256 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-aes-256
+# rsa [1024 | 2048 | 4096]
+p11sak generate-key rsa 1024 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-rsa-1024
+p11sak generate-key rsa 2048 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-rsa-2048
+p11sak generate-key rsa 4096 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-rsa-4096
+# ec [prime256v1 | secp384r1 | secp521r1]
+p11sak generate-key ec prime256v1 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-ec-prime256v1
+p11sak generate-key ec secp384r1 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-ec-secp384r1
+p11sak generate-key ec secp521r1 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-ec-secp521r1
+
+
+echo "** Now list keys and redirect output to pre-files - 'p11sak_test.sh'"
+
+
+# list objects
+p11sak list-key des --slot $SLOT --pin $PKCS11_USER_PIN &> $P11SAK_DES_PRE
```

Comment:
How is checked if the list command did list the expected keys?

Reply:
See below